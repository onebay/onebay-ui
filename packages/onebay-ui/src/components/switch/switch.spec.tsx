import { mount } from '@vue/test-utils'
import Switch from './index'

const factory = (values = {}, slots = { default: ['switch'] }) => {
  return mount(Switch, {
    slots,
    props: { ...values },
  })
}

describe('Switch Snap', () => {
  it('render initial Switch', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Switch -- props title', () => {
    const wrapper = factory({ title: 'title' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Switch -- props checked', () => {
    const wrapper = factory({ title: 'checked', checked: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Switch -- props border', () => {
    const wrapper = factory({ title: 'border', border: false })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Switch -- props disabled', () => {
    const wrapper1 = factory({ title: 'disabled', checked: true, disabled: true })
    expect(wrapper1.element).toMatchSnapshot()
    const wrapper2 = factory({ title: 'disabled', disabled: true })
    expect(wrapper2.element).toMatchSnapshot()
  })

  it('Switch Change checked', async () => {
    const wrapper = factory({ title: 'checked', checked: false })
    await wrapper.setProps({ checked: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('Switch color', () => {
    const wrapper = factory({ checked: true, color: '#ff4455' })
    expect(wrapper.element).toMatchSnapshot()
  })


})

describe('Switch Event', () => {
  it('Switch onChange', () => {
    const mockEvent = {
      target: { checked: false },
      stopPropagation: () => { }
    }
    const wrapper = factory({ title: 'Event' })
    wrapper.find('.ob-switch .ob-switch__switch').trigger('change', { mockEvent })
    expect(wrapper.emitted()).toHaveProperty('change')
    expect(wrapper.vm.isChecked).toBeFalsy()
  })


})
