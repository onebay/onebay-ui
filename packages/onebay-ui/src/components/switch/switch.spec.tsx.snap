// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Snap Switch Change checked 1`] = `
<div
  class="ob-switch"
>
  <div
    class="ob-switch__title"
  >
    checked
  </div>
  <div
    class="ob-switch__container"
  >
    <div
      class="ob-switch__mask"
    />
    <input
      class="ob-switch ob-switch__switch"
      name=""
      type="checkbox"
    />
  </div>
</div>
`;

exports[`Switch Snap Switch color 1`] = `
<div
  class="ob-switch"
>
  <div
    class="ob-switch__title"
  >
    
  </div>
  <div
    class="ob-switch__container"
  >
    <div
      class="ob-switch__mask"
    />
    <input
      class="ob-switch ob-switch__switch"
      name=""
      style="border-color: #ff4455; background-color: rgb(255, 68, 85);"
      type="checkbox"
    />
  </div>
</div>
`;

exports[`Switch Snap render Switch -- props border 1`] = `
<div
  class="ob-switch ob-switch--without-border"
>
  <div
    class="ob-switch__title"
  >
    border
  </div>
  <div
    class="ob-switch__container"
  >
    <div
      class="ob-switch__mask"
    />
    <input
      class="ob-switch ob-switch__switch"
      name=""
      type="checkbox"
    />
  </div>
</div>
`;

exports[`Switch Snap render Switch -- props checked 1`] = `
<div
  class="ob-switch"
>
  <div
    class="ob-switch__title"
  >
    checked
  </div>
  <div
    class="ob-switch__container"
  >
    <div
      class="ob-switch__mask"
    />
    <input
      class="ob-switch ob-switch__switch"
      name=""
      type="checkbox"
    />
  </div>
</div>
`;

exports[`Switch Snap render Switch -- props disabled 1`] = `
<div
  class="ob-switch"
>
  <div
    class="ob-switch__title"
  >
    disabled
  </div>
  <div
    class="ob-switch__container ob-switch--disabled"
  >
    <div
      class="ob-switch__mask"
    />
    <input
      class="ob-switch ob-switch__switch"
      disabled=""
      name=""
      type="checkbox"
    />
  </div>
</div>
`;

exports[`Switch Snap render Switch -- props disabled 2`] = `
<div
  class="ob-switch"
>
  <div
    class="ob-switch__title"
  >
    disabled
  </div>
  <div
    class="ob-switch__container ob-switch--disabled"
  >
    <div
      class="ob-switch__mask"
    />
    <input
      class="ob-switch ob-switch__switch"
      disabled=""
      name=""
      type="checkbox"
    />
  </div>
</div>
`;

exports[`Switch Snap render Switch -- props title 1`] = `
<div
  class="ob-switch"
>
  <div
    class="ob-switch__title"
  >
    title
  </div>
  <div
    class="ob-switch__container"
  >
    <div
      class="ob-switch__mask"
    />
    <input
      class="ob-switch ob-switch__switch"
      name=""
      type="checkbox"
    />
  </div>
</div>
`;

exports[`Switch Snap render initial Switch 1`] = `
<div
  class="ob-switch"
>
  <div
    class="ob-switch__title"
  >
    
  </div>
  <div
    class="ob-switch__container"
  >
    <div
      class="ob-switch__mask"
    />
    <input
      class="ob-switch ob-switch__switch"
      name=""
      type="checkbox"
    />
  </div>
</div>
`;
